src/darwin/Framework/CHIP/MTRBaseDevice.h

```diff
@@ -229,6 +260,38 @@ typedef NS_ENUM(uint8_t, MTRTransportType) {
                           completion:(MTRDeviceResponseHandler)completion
     API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4));
 
+/**
+ * Reads multiple attributes or events from the device.
+ *
+ * Nil value is treated as empty array for attributePaths and eventPaths.
+ *
+ * Lists of attribute and event paths to read can be provided via attributePaths and eventPaths.
+ *
+ * The completion will be called with an error if  the input parameter has a problem(e.g., both attributePaths and eventPaths are
+ * empty.) or the entire read interaction fails. Otherwise it will be called with values, which may be empty (e.g. if no paths
+ * matched the wildcard) or may include per-path errors if particular paths failed.
+ */
+- (void)readWithAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullable)attributePaths
+                    EventPaths:(NSArray<MTREventRequestPath *> * _Nullable)eventPaths
+                        params:(MTRReadParams * _Nullable)params
+                         queue:(dispatch_queue_t)queue
+                    completion:(MTRDeviceResponseHandler)completion MTR_NEWLY_AVAILABLE;
+/**
+ * Subscribes multiple attributes or events from the device.
+ *
+ * Nil value is treated as empty array for attributePaths and eventPaths.
+ *
+ * Lists of attribute and event paths to subscribe can be provided via attributePaths and eventPaths.
+ *
+ * The reportHandler will be called with an error if the input parameter has a problem(e.g., both attributePaths and eventPaths are
+ * empty.).
+ */
+- (void)subscribeWithAttributePaths:(NSArray<MTRAttributeRequestPath *> * _Nullable)attributePaths
+                         EventPaths:(NSArray<MTREventRequestPath *> * _Nullable)eventPaths
+                             params:(MTRSubscribeParams * _Nullable)params
+                              queue:(dispatch_queue_t)queue
+                      reportHandler:(MTRDeviceResponseHandler)reportHandler
+            subscriptionEstablished:(MTRSubscriptionEstablishedHandler _Nullable)subscriptionEstablished MTR_NEWLY_AVAILABLE;
```

Comment:
Should there be a handler here for resubscribe notifications too?

Reply:
I added.